### PR TITLE
Register imageJ agent.

### PR DESCRIPTION
**Pull request by jburel**
This PR sets the correct agent when insight is run as an imageJ plugin.
cc @aleksandra-tarkowska

### CODE REVIEW

**Comment by aleksandra-tarkowska** [comment]
@jburel could you provide more details how that should be tested exactly? I downloaded OMERO.ij from https://ci.openmicroscopy.org/job/OMERO-5.1-merge-build/ICE=3.5,jdk=8_LATEST,label=trout/ but It registered Insight hit `` 2015-09-21 12:28:20.082916+01 |        2``

**Comment by aleksandra-tarkowska** [comment]
unit test would be also worth to add

**Comment by jburel** [comment]
* Install the OMERO.imagej plugin in the  imageJ plugins folder
 * Start imageJ
 * Go to plugins>OMERO>Connect to OMERO
 * Log in
 * Check in DB that we have an entry corresponding to the imageJ agent.

**Comment by aleksandra-tarkowska** [comment]
@jburel that was exactly what I did, still incorrect agent

**Comment by jburel** [comment]
@aleksandra-tarkowska: tested locally  after some adjustment:  "useragent" column in session table is now set to "OMERO.ij"

**Comment by sbesson** [comment]
Closing/reopening to include https://github.com/openmicroscopy/openmicroscopy/pull/4194 in the merge commit for Travis.

**Comment by dominikl** [comment]
Works like expected, useragent name OMERO.ij is used when started up as IJ plugin. Good to merge.

**Comment by jburel** [comment]
@aleksandra-tarkowska, @dominikl: the last commit should now register the correct agent for imageJ (ij), insight (as before), importer.
insight was always used, regardless of the agent set.

**Comment by aleksandra-tarkowska** [comment]
Thank you for setting the correct agent. We are now recording OMERO.ij hit
```
134.36.162.28 - - [22/Sep/2015:13:09:10 +0100] "GET ?version=5.1.3-370-191018b-ice35-b620;os.name=Mac+OS+X;os.arch=x86_64;os.version=10.8.5;java.runtime.version=1.6.0_65-b14-462-11M4609;java.vm.vendor=Apple+Inc. HTTP/1.1" 301 507 "-" "OMERO.ij"
         creation_date         | agent_id 
-------------------------------+----------
 2015-09-22 13:09:10.686175+01 |       10
```

although I have few questions:
 - why another record is recoded when I simply close insight?

 ```
134.36.162.28 - - [22/Sep/2015:13:12:21 +0100] "GET ?version=5.1.3-370-191018b-ice35-b620;os.name=Mac+OS+X;os.arch=x86_64;os.version=10.8.5;java.runtime.version=1.6.0_65-b14-462-11M4609;java.vm.vendor=Apple+Inc. HTTP/1.1" 301 507 "-" "OMERO.ij"
134.36.162.28 - - [22/Sep/2015:13:12:26 +0100] "GET ?version=5.1.3-370-191018b-ice35-b620;os.name=Mac+OS+X;os.arch=x86_64;os.version=10.8.5;java.runtime.version=1.6.0_65-b14-462-11M4609;java.vm.vendor=Apple+Inc. HTTP/1.1" 301 507 "-" "OMERO.ij"

         creation_date         | agent_id 
-------------------------------+----------
 2015-09-22 13:12:26.834753+01 |       10
 2015-09-22 13:12:21.602889+01 |       10
```

**Comment by aleksandra-tarkowska** [comment]
Upgrade check should be called on login just after ``createSession`` and should be omitted on ``joinSession``. This guarantee real stats.

**Comment by jburel** [comment]
Upgrade check now moved to connection phase.
The check when logging out no longer occurs

File: components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 package org.openmicroscopy.shoola.env.data;
 
+import ij.IJ;
```
Comment (sbesson) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java]: Why is this imported in this file?

Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java]: probably left after some debugging...

**Comment by dominikl** [comment]
Tested with a local server, the useragent is set correctly for all clients (OMERO.insight, OMERO.importer, OMERO.ij). But there are still two entries per login, e. g.:
screen shot 2015-09-23 at 09 56 26

**Comment by jburel** [comment]
hummmm

**Comment by dominikl** [comment]
Maybe @aleksandra-tarkowska can double check, in case I just did something wrong.

**Comment by aleksandra-tarkowska** [comment]
@dominikl It is expected to get 2 records in the session table when you first create guest session to check version then acquire user connection. Upgrade check has nothing to do with local database

**Comment by aleksandra-tarkowska** [comment]
Tested using https://ci.openmicroscopy.org/job/OMERO-5.1-merge-build/621/ICE=3.5,jdk=8_LATEST,label=trout/artifact/src/target/OMERO.insight-ij-5.1.3-387-f2a1d2e-ice35-b621.zip

upgrade check was recorded only one on login

```
134.36.162.28 - - [23/Sep/2015:10:50:54 +0100] "GET ?version=5.1.3-387-f2a1d2e-ice35-b621;os.name=Mac+OS... HTTP/1.1" 301 507 "-" "OMERO.ij"

         creation_date         | agent_id 
-------------------------------+----------
 2015-09-23 10:50:54.68814+01  |       10
```

all works as expected.

Small issue noticed that when you switch user insight no longer open after logging in. It was necessary to go and click connect to OMERO again to launch insight. If that is not a blocker, this is good to merge.

**Comment by aleksandra-tarkowska** [comment]
In addition there were also corerct hit recorded for Insight as well as Importer

```
134.36.162.28 - - [23/Sep/2015:11:05:35 +0100] "GET /?version=5.1.3-387-f2a1d2e-ice35-b621;os.name=Mac+OS... HTTP/1.1" 301 503 "-" "OMERO.importer"

134.36.162.28 - - [23/Sep/2015:11:03:09 +0100] "GET /?version=5.1.3-387-f2a1d2e-ice35-b621;os.name=Mac+OS+... HTTP/1.1" 301 503 "-" "OMERO.insight"

         creation_date         | agent_id 
-------------------------------+----------
 2015-09-23 11:05:35.554009+01 |        1
 2015-09-23 11:03:10.234792+01 |        2
```

**Comment by sbesson** [comment]
Thanks @aleksandra-tarkowska, I would suggest we also check registry hits for the various Java clients as part of the 5.1.4-rc1 testing /cc @pwalczysko
Can the issue reported by @aleksandra-tarkowska in https://github.com/openmicroscopy/openmicroscopy/pull/4190#issuecomment-142551005 be fixed elsewhere or is it expected @jburel?

**Comment by jburel** [comment]
We have an old ticket for that https://trac.openmicroscopy.org/ome/ticket/9604
The switch user is more a developer feature.
I will certainly tackle some of the ij issues during 5.2 since the plugin is now used due to workflow improvement

**Comment by jburel** [comment]
--rebased-to #4212

**Pull request merged**